From 81cc24dc74c59b4c6c2ffa9a720ada31b67c44c9 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Tue, 08 Mar 2022 04:40:19 +0200
Subject: [PATCH] [SW-0] rom : fix validation skip check

Fix the validation skip check, which should skip validation of the
images in case boot descriptor initialization fails.

Change-Id: I155b1aa22693976a930661e8c8147010aca0c05a
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index b32a156..d9448d4 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -329,7 +329,7 @@
 			&arcboot_rom_global.boot_ptr_ctx[HL_FW_BOOT_PTR_SPI].load_addr;
 	int ret;
 
-	if (arcboot_rom_boot_desc_init_done(HL_FW_BOOT_PTR_SPI)) {
+	if (!arcboot_rom_boot_desc_init_done(HL_FW_BOOT_PTR_SPI)) {
 		goto skip_bringup;
 	}
 
